src/ui/public/vis/response_handlers/legacy.js

```diff
@@ -17,110 +17,57 @@
  * under the License.
  */
 
-import _ from 'lodash';
-import AggConfigResult from '../../vis/agg_config_result';
 import { VisResponseHandlersRegistryProvider } from '../../registry/vis_response_handlers';
 
-const getSchema = column => {
-  return _.get(column, 'aggConfig.schema.name') || _.get(column, 'aggConfig.schema');
-};
-
-const getType = column => {
-  return _.get(column, 'aggConfig.type.type') || _.get(column, 'aggConfig.type');
-};
-
 const LegacyResponseHandlerProvider = function () {
 
   return {
     name: 'legacy',
-    handler: function (table) {
+    handler: function (table, dimensions) {
       return new Promise((resolve) => {
         const converted = { tables: [] };
 
-        // check if there are buckets after the first metric
-        const metricsAtAllLevels = table.columns.findIndex(column => getType(column) === 'metrics') <
-          _.findLastIndex(table.columns, column => getType(column) === 'buckets');
-
-        const splitColumn = table.columns.find(column => getSchema(column) === 'split');
-        const numberOfMetrics = table.columns.filter(column => getType(column) === 'metrics').length;
-        const numberOfBuckets = table.columns.filter(column => getType(column) === 'buckets').length;
-        const metricsPerBucket = numberOfMetrics / numberOfBuckets;
+        const split = (dimensions.splitColumn || dimensions.splitRow);
 
-        if (splitColumn) {
-          const splitAgg = splitColumn.aggConfig;
+        if (split) {
+          converted.direction = dimensions.splitRow ? 'row' : 'column';
+          const splitColumnIndex = split[0].accessor;
```

Comment:
question: If I understand this correctly, `splitColumnIndex` could possibly contain the index of the split row -- is that expected? Or is this something we need to even be concerned about?

Here's what I mean:
```javascript
const dimensions = { splitRow: [{ accessor: 1 }] };
const split = (dimensions.splitColumn || dimensions.splitRow); // split = [{ accessor: 1 }]
const splitColumnIndex = split[0].accessor; // 1, which is actually the splitRow accessor
```

Reply:
yeah, exactly. the naming is not the best i guess :dagger: 
so ... split = splitColumn || splitRow .... we have our split now, we don't care in which direction we need to split

so we have our split somewhere in our datatable.
now split_columnIndex just tells us which column in the datatable is producing the split, and has nothing to do with split direction. 

(like columnIndex, rowIndex in raw data .... it indicates position inside the table, where the split direction tells us how to render the whole thing)

hope this makes it clearer.